source/developer/oauth2-applications.rst

```diff
@@ -0,0 +1,80 @@
+
+5 - (Optional) If you would like to allow users on your system who are not System Admins to create OAuth 2.0 applications, then set `Restrict managing integrations to Admins <https://docs.mattermost.com/administration/config-settings.html#restrict-managing-integrations-to-admins>`_ to **False**.
+
+Register an OAuth 2.0 Application
+~~~~~~~~~~~~~~~~~~~~~~
+1 - Go to **Main Menu** > **Integrations**
+
+2 - Click **OAuth 2.0 Applications**, then click **Add OAuth 2.0 Application**
+
+3 - Set **Is Trusted**: When set to **Yes**, your application is considered trusted by Mattermost. This means Mattermost doesn't require users to accept authorization when signing to third-party applications. When set to **No**, users will be provided with the following page to accept or deny authorization when authenticating for the first time.
+
+.. image:: ../images/oauth2_authorization_screen.PNG
+
+4 - Specify the **Display Name**: Enter a name for your application made of up to 64 characters. This is the name users will see when granting access to the application, when viewing a list of authorized applications in **Account Settings > Security > OAuth 2.0 Applications** and when viewing a list of OAuth 2.0 applications in **Main Menu** > **Integrations**.
+
+5 - Add **Description**: This is a short description of your application that users will see when viewing a list of authorized applications in **Account Settings > Security > OAuth 2.0 Applications**.
+
+6 - Specify the **Homepage**: This is the homepage of the OAuth 2.0 application. The URL must be a valid URL and start with ``http://`` or ``https://`` depending on your server configuration.
+
+7 - (Optional) Add **Icon URL**: The image users will see when viewing a list of authorized applications in **Account Settings > Security > OAuth 2.0 Applications** and when viewing a list of OAuth 2.0 applications in **Main Menu** > **Integrations**. Must be a valid URL and start with ``http://`` or ``https://``.
+
+8 - Add **Callback URLs**: This are the URL(s) to which Mattermost will redirect users after accepting or denying authorization of your application, and which will handle authorization codes or access tokens. Each URL must be on a separate line and start with ``http://`` or ``https://``.
+
+9 - Click **Save** to create the application. 
+
+.. image:: ../images/oauth2_app_screen.png
+
+10 - You will be provided with a **Client ID**, **Client Secret** and the authorized redirect URLs. Save these values and use them in your application to connect it to Mattermost.
+
+.. image:: ../images/oauth2_confirmation_screen.PNG
+
+Note: **Client Secret** can be regenerated by the application creator or System Administrator, preventing new users from accessing the OAuth 2.0 application until the client secret has been updated in your app configuration.
```

Comment:
actually the tokens created with an old secret are still valid. If you regenerate the secret new authorizations will fail, also the request for a refresh token will fail unless the secret is updated in the OAuth2 app